lib/auth/webauthn/login_test.go

```diff
@@ -860,6 +870,157 @@ func TestLogin_scopeAndReuse(t *testing.T) {
 	})
 }
 
+func TestLoginFlow_userVerification(t *testing.T) {
+	// Prepare a user and a pair of registered devices.
+	mfaDev, err := mocku2f.Create()
+	require.NoError(t, err)
+
+	pwdlessDev, err := mocku2f.Create()
+	require.NoError(t, err)
+	pwdlessDev.IgnoreAllowedCredentials = true // passwordless settings
+	pwdlessDev.SetUV = true
+	pwdlessDev.AllowResidentKey = true
+
+	const user = "llama"
+	const origin = "https://example.com"
+	webIdentity := newFakeIdentity(user)
+	webConfig := &types.Webauthn{RPID: "example.com"}
+
+	ctx := context.Background()
+	register := func(t *testing.T, dev *mocku2f.Key, rr wanlib.RegisterResponse) {
+		webRegistration := &wanlib.RegistrationFlow{
+			Webauthn: webConfig,
+			Identity: webIdentity,
+		}
+
+		cc, err := webRegistration.Begin(ctx, rr.User, rr.Passwordless)
+		require.NoError(t, err)
+
+		ccr, err := dev.SignCredentialCreation(origin, cc)
+		require.NoError(t, err)
+		rr.CreationResponse = ccr
+
+		_, err = webRegistration.Finish(ctx, rr)
+		require.NoError(t, err)
+	}
+
+	// Register devices. They are "persisted" in the fake identify.
+	register(t, mfaDev, wanlib.RegisterResponse{
+		User:       user,
+		DeviceName: "mfa",
+	})
+	register(t, pwdlessDev, wanlib.RegisterResponse{
+		User:         user,
+		DeviceName:   "pwdless",
+		Passwordless: true,
+	})
+
+	tests := []struct {
+		name                      string
+		exts, requiredExts        *mfav1.ChallengeExtensions
+		dev                       *mocku2f.Key
+		wantAssertionVerification string
+		wantErr                   string
+	}{
```

Comment:
Do I correctly assume that some of these tests are for cases not explicitly affected by the PR, but they are here for completeness?

Reply:
They are all for the changes in the PR. Which ones did you think aren't?

Reply:
It's because you used `mfav1.ChallengeScope_CHALLENGE_SCOPE_ADMIN_ACTION`, which gave me an idea that these are completely different cases.

Reply:
No, any scope would do. I would have used PASSWORD_CHANGE (which makes for a more realistic test), but we don't have it defined yet.